lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -94,29 +99,251 @@ class IntProgCmdProvider: public CmdProvider {
+     */
+    virtual bool next(const CommandInfo*& cmd, const ResponseInfo*& rsp)
+    {
+        /*Keep original command.*/
+        const CommandInfo* originalCmd = &m_cmds[nextCmd];
+
+        /*Copy original command.*/
+        CommandInfo updatedCmd = *originalCmd;
+
+        /*Update the command with required content to write in the flash memory.*/
+        updateFwProgCmd(&updatedCmd);
+
+        /*Save the updated command information.*/
+        m_updatedCmd = updatedCmd;
+
+        cmd = &m_updatedCmd;
+
+        if(nullptr != cmd)
+        {   
+            rsp = &m_responses[nextCmd];
+
+            /*Move to the next chunk.*/
+
+            if (nextCmd == FLSPRG_SET_ADDR)
+            {
+                nextCmd = FLSPRG_WRITE_BLOCK;
+            }
+            else
+            {
+                nextCmd = FLSPRG_SET_ADDR;
+            }
+
+            return true;
+        }
+        else
+        {
+            rsp = nullptr;
+            return false;
+        }
+    }
+
+/**
+ * Get the command type.
+ * @return The command type, which is FW_PROG for firmware programming.
+ */
+CMD_TYPE getCmdType() const override 
+{
+    return FW_PROG;
 }
-#endif
+
+    FileManager m_fileManager; /**< Instance of Filemanger class.*/
+
+private:
+    static  CommandInfo m_cmds[]; /**< Array of command information.*/
+    static  ResponseInfo m_responses[]; /**< Array of response information.*/
+    const char* m_fileName = "/zumo_firmware.bin"; /**< Name of the file to be read.*/
+
+    /**
+     * @brief Enumeration of programming commands.
+     * This enumeration defines the various programming commands used during firmware programming.
+     * It includes commands for setting memory addresses and writing data blocks.
+     */
+   enum ProgCmds {
+    FLSPRG_SET_ADDR,    /**< Command to set the memory address. */
+    FLSPRG_WRITE_BLOCK  /**< Command to write a data block to memory. */
+};
+
+    /** 
+    * @brief The next programming command to be executed. 
+    * This variable represents the next programming command to be executed in the sequence.
+    * It initializes with FLSPRG_SET_ADDR, typically used to set the memory address first.
+    */
+    ProgCmds nextCmd = FLSPRG_SET_ADDR;
+    CommandInfo m_updatedCmd; /**<Intermediate storage for updated commands.*/
+    uint8_t* m_updatedCmdBuffer; /**<Buffer for updated commands.*/
+    size_t m_updatedCmdBufferSize; /**<Size of the buffer.*/
+
+    /**
+     * @brief Current memory address for writing data during firmware programming. 
+     * This variable represents the current memory address where data will be written
+     * during the firmware programming process. It is used to keep track of the
+     * memory address for sequential data writes.
+     */
+    uint16_t  m_currWriteMemAddr; 
+
+   /**
+    * @brief Updates the command according to specific requirements.
+    *
+    * This method updates the provided command based on specific requirements

```

Comment:
I don't see any requirements it is basefd on.  Just comment what it does and why.

Reply:
It will be take care by the next commit.